packages/unmock-fetch/src/index.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  ISerializedRequest,
+  ISerializedResponse,
+  OnSerializedRequest,
+} from "unmock-core/dist/interfaces";
+import buildFetch from "./fetch";
+export { buildFetch };
+
+export type Listener = (
+  req: ISerializedRequest,
+  respond: (res: ISerializedResponse) => void,
+) => void;
+
+export type Fetch = (url: RequestInfo, init?: RequestInit) => Promise<Response>;
+
+// tslint:disable-next-line:no-namespace
+declare namespace global {
+  let fetch: Fetch;
+}
+
+// tslint:disable-next-line:no-namespace
+declare namespace window {
+  let fetch: Fetch;
+}
+
+let mitm: Mitm | undefined;
```

Comment:
Is this the same `mitm` we have in the backend?

Reply:
I see that, but it seems you tried to maintain the same interface, hence the question. It's a bit misleading IMO to have two `Mitm` classes with different purposes.

Reply:
Yeah I can rename it if it makes development simpler for us. But in general users of this would not know anything about `node-mitm` and its API so I just picked a short name that makes it clear what it does. I can call it `FetchInterceptor` or something.